source/common/config/api_type_oracle.cc

```diff
@@ -5,13 +5,27 @@
 namespace Envoy {
 namespace Config {
 
+// To exclude rejecting below v2 protos.
+static const std::set<absl::string_view>& excludeV2ProtoSet() {
+  CONSTRUCT_ON_FIRST_USE(
+      std::set<absl::string_view>,
+      {"envoy.config.health_checker.redis.v2", "envoy.config.filter.thrift.router.v2alpha1",
+       "envoy.config.resource_monitor.fixed_heap.v2alpha",
+       "envoy.config.resource_monitor.injected_resource.v2alpha",
+       "envoy.config.retry.omit_canary_hosts.v2", "envoy.config.retry.previous_hosts.v2"});
+}
+
 const Protobuf::Descriptor*
 ApiTypeOracle::getEarlierVersionDescriptor(const std::string& message_type) {
   const auto previous_message_string = getEarlierVersionMessageTypeName(message_type);
   if (previous_message_string != absl::nullopt) {
     const Protobuf::Descriptor* earlier_desc =
         Protobuf::DescriptorPool::generated_pool()->FindMessageTypeByName(
             previous_message_string.value());
+    auto itr = excludeV2ProtoSet().find(earlier_desc->full_name());
```

Comment:
Sorry, I realized in https://github.com/envoyproxy/envoy/issues/14735#issuecomment-761930629 that this won't work as intended; we need to always allow v2 --> v3 upgrade, we need to update the code of all the extensions you list above to work with v3 config protos and we need to move the denylist to the exception reporting code. 

I'm going to work on a PR to do this I think, since we need to make this land for v3 and I'm fairly familiar with the code. Thanks for all the work you've done to get us to this point, it's provided useful exploration of what the underlying issues are.

Reply:
oh ok thanks @htuch for your kind help. please suggest if this issue can ben closed now ? i can now focus on another pending work of community :)